File: 백준/Bronze/22864. 피로도/피로도.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int a = Integer.parseInt(input[0]);
        int b = Integer.parseInt(input[1]);
        int c = Integer.parseInt(input[2]);
        int m = Integer.parseInt(input[3]);
        int work = 0;
        int fatigue = 0;
        for(int i=0;i<24;i++){
            if(fatigue+a <=m){
                work+=b;
                fatigue+=a;
            }else{
                fatigue = Math.max(0,fatigue-c);
            }
        }
        System.out.println(work);
        }
    }

File: 백준/Bronze/22864. 피로도/피로도.py
import sys
input = sys.stdin.readline

a, b, c, m = map(int, input().split())
work = 0
fatigue = 0

for _ in range(24):
    if fatigue + a <= m:
        fatigue += a
        work += b
    else:
        fatigue = max(0, fatigue - c)

print(work)
